tests/bdd/test_replicas_garbage_collection.py

```diff
@@ -0,0 +1,147 @@
+    # When starting Mayastor instances with the deployer a bind mount is created from /tmp to
+    # /host/tmp, so create disk images in /tmp
+    for disk in [POOL_DISK1, POOL_DISK2]:
+        path = "/tmp/{}".format(disk)
+        if os.path.exists(path):
+            os.remove(path)
+        with open(path, "w") as file:
+            file.truncate(20 * 1024 * 1024)
+
+
+# This fixture will be automatically used by all tests.
+# It starts the deployer which launches all the necessary containers.
+# A pool is created for convenience such that it is available for use by the tests.
+@pytest.fixture(autouse=True)
+def init():
+    create_pool_disk_images()
+
+    # Shorten the reconcile periods and cache period to speed up the tests.
+    common.deployer_start_with_args(
+        [
+            "-j",
+            "-m",
+            "2",
+            "-w",
+            "10s",
+            "--reconcile-idle-period",
+            "1s",
+            "--reconcile-period",
+            "1s",
+            "--cache-period",
+            "1s",
+        ]
+    )
+
+    # Create pools
+    common.get_pools_api().put_node_pool(
+        MAYASTOR_1,
+        POOL1_UUID,
+        CreatePoolBody(["aio:///host/tmp/{}".format(POOL_DISK1)]),
+    )
+    common.get_pools_api().put_node_pool(
+        MAYASTOR_2,
+        POOL2_UUID,
+        CreatePoolBody(["aio:///host/tmp/{}".format(POOL_DISK2)]),
+    )
+
+    # Create and publish a volume on node 1
+    cfg = common.get_cfg()
+    request = CreateVolumeBody(
+        VolumePolicy(False), NUM_VOLUME_REPLICAS, VOLUME_SIZE, _configuration=cfg
+    )
+    common.get_volumes_api().put_volume(VOLUME_UUID, request)
+    common.get_volumes_api().put_volume_target(
+        VOLUME_UUID, MAYASTOR_1, Protocol("nvmf")
+    )
+
+    yield
+    common.deployer_stop()
+
+
+@scenario(
+    "features/garbage-collection/replicas.feature", "destroying an orphaned replica"
+)
+def test_destroying_an_orphaned_replica():
+    """destroying an orphaned replica."""
+
+
+@given("a replica which is managed but does not have any owners")
+def a_replica_which_is_managed_but_does_not_have_any_owners():
+    """a replica which is managed but does not have any owners."""
+
+    # Kill the Mayastor instance which does not host the nexus.
+    common.kill_container(MAYASTOR_2)
+
+    # Attempt to delete the volume. This will leave a replica behind on the node that is
+    # inaccessible.
+    try:
+        common.get_volumes_api().del_volume(VOLUME_UUID)
+    except Exception as e:
+        # A Mayastor node is inaccessible, so deleting the volume will fail because the replica
+        # on this node cannot be destroyed. Attempting to do so results in a timeout. This is
+        # expected and results in a replica being orphaned.
+        exception_info = e.__dict__
+        assert exception_info["status"] == requests.codes["request_timeout"]
+        pass
+
+    # Allow some time for the replica to be disowned.
+    time.sleep(5)
```

Comment:
should this be a retry loop with small sleeps but a larger timeout to avoid failing if it ever takes longer for whatever reason?